### PR TITLE
Removing individual queue settings from SettingsHolder

### PR DESCRIPTION
**Pull request by SeanFeldman**
Connects to #282 

First step in series of changes to come, required to remove ReadOnlySettings usage at runtime.
This is a WIP.
- [x] Both topologies expose `TopologySettings`. 
- `TopologySettings` added to provide
  - [x] `QueueSettings` 
  - [x] `TopicSettings`
  - [x] `SubscriptionSettings`
- Factory extensions obsoleted as they no longer serve the purpose where needed for.

Separate PRs:
- Obsolete on queue/topic/subscription conditional `ForwardDeadLetteredMessagesTo` and `EnableExpress` as there's no more need for those (`TODO` comments left in place)
- `AzureServiceBusTransportExtensions` - anything other that topology extension methods should "hang off" the topology extensions. I.e. topology should be specified first, then the rest of the queues/topic/subscription settings.
- Other configuration settings to follow this pattern

We paired with @boblangley to spike this initial attempt. I presume this is the direction we want to go with.

### CODE REVIEW

File: src/Tests/Receiving/When_comparing_performance_for_prefetch.cs [hunk 1]
```diff
@@ -24,6 +24,7 @@ public class When_comparing_performance_for_prefetch
             var cts = new CancellationTokenSource(TimeSpan.FromSeconds(60));
             // default settings
             var settings = new DefaultConfigurationValues().Apply(new SettingsHolder());
+            settings.Set<TopologySettings>(new TopologySettings());
```
Comment (danielmarbach) [inline src/Tests/Receiving/When_comparing_performance_for_prefetch.cs]: Why does this need to be in settings?

File: src/Tests/API/APIApprovals.ApproveAzureServiceBusTransport.approved.txt [hunk 2]
```diff
@@ -91,7 +91,7 @@
     {
         public NServiceBus.AzureServiceBusQueueSettings AutoDeleteOnIdle(System.TimeSpan autoDeleteOnIdle) { }
         public NServiceBus.AzureServiceBusQueueSettings DefaultMessageTimeToLive(System.TimeSpan defaultMessageTimeToLive) { }
-        public NServiceBus.AzureServiceBusQueueSettings DescriptionFactory(System.Func<string, NServiceBus.Settings.ReadOnlySettings, Microsoft.ServiceBus.Messaging.QueueDescription> factory) { }
+        public NServiceBus.AzureServiceBusQueueSettings DescriptionFactory(System.Action<Microsoft.ServiceBus.Messaging.QueueDescription> factory) { }
```
Comment (danielmarbach) [inline src/Tests/API/APIApprovals.ApproveAzureServiceBusTransport.approved.txt]: So this is no longer a factory but a configurer or modifier?

**Comment by danielmarbach** [comment]
Like where this is moving. Bit sceptic about the depth of the settings but let's stick to this we can always tear it apart again.

File: src/Tests/Receiving/When_comparing_performance_for_prefetch.cs (hunk 1 above)
Comment (SeanFeldman) [inline src/Tests/Receiving/When_comparing_performance_for_prefetch.cs]: It doesn't.

**Comment by SeanFeldman** [comment]
@Particular/azure-maintainers please review

**Comment by SeanFeldman** [comment]
@Particular/azure-maintainers please review

**Comment by boblangley** [comment]
Merged #451

**Pull request merged**